### PR TITLE
feat(payments-next): Sign-in/sign-up form

### PR DESCRIPTION
**Pull request by xlisachan**
## This pull request

- [x] Create a client component for the Radix form containing the email input, checkbox, and Continue button
- [x] Add to Checkout page
- [x] Add "Continue with Google" and "Continue with Apple" buttons
- [x] Clean up ftl ids in en.ftl

## Issue that this pull request solves

Closes: FXA-7580

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)
Mobile
Screenshot 2024-10-17 at 3 16 02 PM

Tablet
Screenshot 2024-10-17 at 3 16 20 PM

Desktop
Screenshot 2024-10-17 at 3 19 38 PM

### CODE REVIEW

**Review by david1alvarez — APPROVED** [review]

File: apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/start/page.tsx [hunk 1]
```diff
@@ -46,74 +52,37 @@ export default async function Checkout({ params }: { params: CheckoutParams }) {
...
-            </form>
+          <h3 className="font-semibold text-grey-600 text-start">
+            {l10n.getString(
+              'checkout-create-account',
+              'Create a Mozilla account'
+            )}
+          </h3>
+
+          <div className="flex flex-col gap-4 mt-6 mb-10 desktop:flex-row desktop:items-center desktop:justify-center">
+            <button className="w-full flex items-center justify-center bg-[#F9F9FA] border border-grey-300 font-medium rounded text-sm">
```
Comment (StaberindeZA) [inline apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/start/page.tsx:72]: Could this be updated to use the `BaseButton` component instead with a new variant?

These buttons also seem to be a different size than the Figma file. Could it be changed to conform to what's in Figma?

Comment (xlisachan) [reply apps/payments/next/app/[locale]/[offeringId]/checkout/[interval]/[cartId]/start/page.tsx:72]: I'll make an update to use the `BaseButton` component. As discussed, I'll change them to conform to how these buttons appear in FxA.

**Review by StaberindeZA — APPROVED** [review]
r+. TY for the additional changes

**Comment by xlisachan** [comment]
@mozilla/fxa-l10n , hi! This PR creates new strings/re-uses old ones using the `next-` prefix; no changes to localization or exposing strings. Thanks!

File: libs/payments/ui/src/lib/client/components/SignInForm/en.ftl [hunk 2]
```diff
@@ -0,0 +1,7 @@
+continue = Continue
+checkout-enter-your-email = Enter your email
```
Comment (bcolsson) [inline libs/payments/ui/src/lib/client/components/SignInForm/en.ftl:2]: I know these aren't technically landing in front of localizers yet, but I'd recommend namespacing these.

**Review by bcolsson — CHANGES_REQUESTED** [review]

File: libs/payments/ui/src/lib/client/components/SignInForm/en.ftl (hunk 2 above)
Comment (xlisachan) [reply libs/payments/ui/src/lib/client/components/SignInForm/en.ftl]: Thanks for the feedback, @bcolsson! I've updated the ids as follows: 
```
signin-form-continue-button = Continue
signin-form-email-input = Enter your email
```

**Review by bcolsson — APPROVED** [review]

**Pull request merged**